libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -367,9 +367,6 @@ private ModuleEntitlements getModuleScopeEntitlements(
      * @return true if permission is granted regardless of the entitlement
      */
     boolean isTriviallyAllowed(Class<?> requestingClass) {
-        if (generalLogger.isTraceEnabled()) {
```

Comment:
logging in entitlements can be sketchy, especially on such a critical path. this breaks in a settings test that sets log level to trace

Reply:
The trouble to me isn't the test, but (file) logging itself triggering further entitlement checks, especially on this path.

Reply:
If you insist, we can move the stacktrace down into 
```
generalLogger.trace("Entitlement not trivially allowed");
```
once we've granted trusted system classes

Reply:
Yeah your scenario is a bit concerning. I'm good with moving it down if you are.

Reply:
(Oh and if you feel like you want to delete it, I can live with that too. I was just hesitating because being unable to log is a big problem for a system as complex as entitlements, so we just want to take some care before we resign ourselves to having "no logging zones" in the code.)